### DISCUSSION TITLE
When I use setError, the message shows up on page but seems to not set on the formState errors

### DISCUSSION THREAD

**Discussion by AmodeusR** [react-hook-form/react-hook-form#8959] [Q&A]
It's quite an odd behavior, because the messages shows up, but since it doesn't go to the formState Errors, the fields never activate the error styles which are based on the error field being non-empty.

It's a little bit tricky to explain in words, so I made a video to show what's happening. I demonstrate how the error message shows up when the passwords are different, but only that. After that I show how should it looks like, with the red borders and such. The undefined shows how the setError is not actually setting the formState errors message, but instead doing some kind of witchery, since the borders and messages are activated when the errors are not undefined, but the message still shows up without the border through the setError.

**Comment by Moshyfawn** [comment]
Can u, please, upload the video as is in your description instead of posting a link? Just trying to avoid any harmful links. Ideally, a minimal reproduction codesandbox as well, ofc.
